contentcuration/search/management/commands/set_contentnode_tsvectors.py

```diff
@@ -26,34 +28,40 @@ class Command(BaseCommand):
     def handle(self, *args, **options):
         start = time.time()
 
-        tsvector_not_already_inserted_query = ~Exists(ContentNodeFullTextSearch.objects.filter(contentnode_id=OuterRef("id")))
+        all_published_channels = list(Channel.objects.filter(main_tree__published=True, deleted=False).values("id", "main_tree__tree_id"))
 
-        tsvector_node_query = (ContentNode._annotate_channel_id(ContentNode.objects)
-                               .annotate(contentnode_tags=StringAgg("tags__tag_name", delimiter=" "),
-                                         keywords_tsvector=CONTENTNODE_KEYWORDS_TSVECTOR,
-                                         author_tsvector=CONTENTNODE_AUTHOR_TSVECTOR)
-                               .filter(tsvector_not_already_inserted_query, published=True, channel_id__isnull=False)
-                               .values("id", "channel_id", "keywords_tsvector", "author_tsvector").order_by())
-
-        insertable_nodes_tsvector = list(tsvector_node_query[:CHUNKSIZE])
         total_tsvectors_inserted = 0
 
-        while insertable_nodes_tsvector:
-            logging.info("Inserting contentnode tsvectors.")
+        for published_channel in all_published_channels:
+            tsvector_not_already_inserted_query = ~Exists(ContentNodeFullTextSearch.objects.filter(contentnode_id=OuterRef("id")))
+            tsvector_nodes_query = (ContentNode.objects
+                                    .annotate(channel_id=Value(published_channel["id"]),
+                                              contentnode_tags=StringAgg("tags__tag_name", delimiter=" "),
+                                              keywords_tsvector=CONTENTNODE_KEYWORDS_TSVECTOR,
+                                              author_tsvector=CONTENTNODE_AUTHOR_TSVECTOR)
+                                    .filter(tsvector_not_already_inserted_query, tree_id=published_channel["main_tree__tree_id"], published=True, complete=True)
```

Comment:
A small flag - I've added a filter that only creates tsvectors for published and complete nodes. This is how we filter for tsvector contentnodes during channel publishing so it feels like during initial creation of tsvectors we should follow the same filters. What do you think? cc @rtibbles @bjester.